helix-term/src/commands.rs

```diff
@@ -780,6 +783,58 @@ fn replace(cx: &mut Context) {
     })
 }
 
+fn switch_case(cx: &mut Context) {
+    let (view, doc) = current!(cx.editor);
+    let transaction =
+        Transaction::change_by_selection(doc.text(), doc.selection(view.id), |range| {
+            let text: Tendril = range
+                .fragment(doc.text().slice(..))
+                .chars()
+                .map(|ch| {
+                    if ch.is_lowercase() {
+                        ch.to_uppercase().collect()
```

Comment:
Ah, these methods return an iterator, hmm. Maybe we can use `flat_map` and directly process these?

Reply:
Ah yes, good point.

Reply:
So with `flat_map` you should be able to return these iterators directly without `.collect()`. Might be a problem since all the branches return a different type though?

Reply:
I must admit; this isn't my area of expertise, but I don't see much of an alternative. Or at-least no ones which also allocate. 
With the exception of a custom enum combining the three branches, which also implements Iterator.

Any other suggestion?

Reply:
I think the current implementation is fine for now :)